File: src/main/java/lesson6/Dictionary.java
package lesson6;

import java.util.*;

public class Dictionary {
    private final Map<String, Set<String>> map = new HashMap<>();
    public void add(String phone, String name) {
        Set<String> record = map.computeIfAbsent(name, k -> new HashSet<String>());
        record.add(phone);
    }

    public Set<String> get(String name) {
        return map.get(name);
    }
}

File: src/main/kotlin/lesson6_kt/Dictionary.kt
package lesson6_kt

class Dictionary {
    private val map = HashMap<String, HashSet<String>>()

    fun add(phone : String, name : String) {
        val record = map.getOrPut(name) { HashSet<String>() }
        record.add(phone)
    }

    fun get(name : String) : Set<String>? {
        return map[name]
    }
}